src/app/(sidebar)/account/fund/page.tsx

```diff
@@ -1,5 +1,134 @@
 "use client";
 
+import { useEffect, useState } from "react";
+import { Alert, Card, Input, Text, Button } from "@stellar/design-system";
+
+import { shortenStellarAddress } from "@/helpers/shortenStellarAddress";
+import { validatePublicKey } from "@/helpers/validatePublicKey";
+import { useFriendBot } from "@/query/useFriendBot";
+import { useStore } from "@/store/useStore";
+
+import "../styles.scss";
+
 export default function FundAccount() {
-  return <div>Fund Account</div>;
+  const { account, network } = useStore();
+
+  const [showAlert, setShowAlert] = useState<boolean>(false);
+  const [generatedPublicKey, setGeneratedPublicKey] = useState<string>("");
+  const [inlineErrorMessage, setInlineErrorMessage] = useState<string | null>(
+    null,
+  );
+
+  const { error, isError, isLoading, isSuccess, refetch, isFetchedAfterMount } =
+    useFriendBot({
+      network: network.id,
+      publicKey: generatedPublicKey,
+    });
+
+  useEffect(() => {
+    if (isError || isSuccess) {
+      setShowAlert(true);
+    }
+  }, [isError, isSuccess]);
+
+  useEffect(() => {
+    if (generatedPublicKey) {
```

Comment:
needed for when user clicks `Fill in with generated key` after inputting the wrong string type in `onChange`